BuoyTest/App.js

```diff
@@ -0,0 +1,39 @@
+import React, {Component} from 'react';
+import { Platform, StyleSheet, Text, View, StatusBar } from 'react-native';
+import { Provider } from 'react-redux';
+import { createStore, applyMiddleware } from 'redux';
+import ReduxThunk from 'redux-thunk';
+import reducers from './src/reducers';
+import AppRoot from './AppRoot';
+import {
+  CocktailsList,
+  CocktailDetails
+} from './src/components/cocktails';
+import { createStackNavigator, createAppContainer } from 'react-navigation';
+
+class App extends Component {
+  render() {
+    const store = createStore(reducers, {}, applyMiddleware(ReduxThunk));
```

Comment:
Jero, the creation of the store must be outside a render function. Actually, it should be outside of a file that contains React components. 
Renders are executed several times, so this could cause performance issues. Please see how is the component lifecycle here https://code.likeagirl.io/understanding-react-component-life-cycle-49bf4b8674de

Reply:
Updated! There is a new Store.js file to create the store.